src/Microsoft.Extensions.ServiceDiscovery.Dns/DnsSrvServiceEndPointResolver.cs

```diff
@@ -53,20 +53,29 @@ protected override async Task ResolveAsyncCore()
         var srvRecords = result.Answers.OfType<SrvRecord>();
         foreach (var record in srvRecords)
         {
-            if (!lookupMapping.TryGetValue(record.Target, out var targetRecord))
+            if (lookupMapping.TryGetValue(record.Target, out var targetRecord))
             {
-                continue;
+                ttl = MinTtl(record, ttl);
+                if (targetRecord is AddressRecord addressRecord)
+                {
+                    AddEndpoint(endPoints, record, addressRecord);
+                    continue;
+                }
```

Comment:
Correct me if I'm misunderstandning the code, but the logic in this method seems to only register the last `IPEndpoint` from the additional section that belongs to a given `SrvRecord`. (`lookupMapping[record.DomainName] = record` overwrites the previous ones.)

On the other hand, if there are no matching address records in additional section, the new lookup logic will add all the endpoints resolved.

Would it make sense to harmonize the two cases and always add all the Address endpoints that associated with an `SrvRecord`?